From d5e82cd0b6076f33b86e0285ef1c0dba8a14112e Mon Sep 17 00:00:00 2001
From: Ahmad Fatoum <a.fatoum@pengutronix.de>
Date: Thu, 9 Jul 2020 13:00:16 +0200
Subject: [PATCH] meson.build: do not hardcode 'linux' as the host system

OE build system can set this to other values that include 'linux',
e.g. 'linux-gnueabi'. This led to glib always being built without
libmount, mkostemp and selinux support.

Upstream-Status: Inappropriate [other]
Signed-off-by: Ahmad Fatoum <a.fatoum@pengutronix.de>
---
 meson.build | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/meson.build b/meson.build
index dd95c750b5ea..8bcacaf3c7e1 100644
--- a/meson.build
+++ b/meson.build
@@ -604,7 +604,7 @@ else
 endif
 message('Checking whether to use statfs or statvfs .. ' + stat_func_to_use)
 
-if host_system == 'linux'
+if host_system.contains('linux')
   if cc.has_function('mkostemp',
                      prefix: '''#define _GNU_SOURCE
                                 #include <stdlib.h>''')
@@ -1810,7 +1810,7 @@ glib_conf.set_quoted('GLIB_LOCALE_DIR', join_paths(glib_datadir, 'locale'))
 # libmount is only used by gio, but we need to fetch the libs to generate the
 # pkg-config file below
 libmount_dep = []
-if host_system == 'linux' and get_option('libmount')
+if host_system.contains('linux') and get_option('libmount')
   libmount_dep = [dependency('mount', version : '>=2.23', required : true)]
   glib_conf.set('HAVE_LIBMOUNT', 1)
 endif
@@ -1820,7 +1820,7 @@ if host_system == 'windows'
 endif
 
 selinux_dep = []
-if host_system == 'linux'
+if host_system.contains('linux')
   selinux_dep = dependency('libselinux', required: get_option('selinux'))
 
   glib_conf.set('HAVE_SELINUX', selinux_dep.found())
-- 
2.27.0

